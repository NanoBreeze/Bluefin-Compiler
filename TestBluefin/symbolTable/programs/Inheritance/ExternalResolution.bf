/*
When we use derived class, we expect to be able to access the base class' fields and member methods.
*/

struct Base {
	int a;
	float b;
	void f() {}
};

struct Der extends Base {
	void g() {}
};

struct DerDer extends Der {
	float h() {	return 3; }
};

int x;

int main() {
	DerDer derder;
	derder.a;
	derder.b;
	derder.f();
	derder.g();
	derder.h();
	
	derder.x;
}