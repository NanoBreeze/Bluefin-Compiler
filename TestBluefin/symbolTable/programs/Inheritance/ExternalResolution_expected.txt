declare - Base - c_struct - t_Base
setCurrentScope - Level 1
declare - a - c_var - t_int
declare - b - c_var - t_float
declare - f - c_function - t_void
enterScope - Level 2
exitScope - Level 2
exitScope - Level 1
resolve - Base - c_struct - t_Base
declare - Der - c_struct - t_Der
setCurrentScope - Level 1
declare - g - c_function - t_void
enterScope - Level 2
resolve - Base::g - UNRESOLVED
exitScope - Level 2
exitScope - Level 1
resolve - Der - c_struct - t_Der
declare - DerDer - c_struct - t_DerDer
setCurrentScope - Level 1
declare - h - c_function - t_float
enterScope - Level 2
resolve - Der::h - UNRESOLVED
exitScope - Level 2
exitScope - Level 1
declare - x - c_var - t_int
declare - main - c_function - t_int
enterScope - Level 1
declare - derder - c_var - t_DerDer
exitScope - Level 1
resolve - int - c_builtinType - t_int
resolve - float - c_builtinType - t_float
resolve - void - c_builtinType - t_void
resolve - void - c_builtinType - t_void
resolve - float - c_builtinType - t_float
resolve - int - c_builtinType - t_int
resolve - int - c_builtinType - t_int
resolve - DerDer - c_struct - t_DerDer
resolve - derder - c_var - t_DerDer
resolve - DerDer::a - c_var - t_int
resolve - derder - c_var - t_DerDer
resolve - DerDer::b - c_var - t_float
resolve - derder - c_var - t_DerDer
resolve - DerDer::f - c_function - t_void
resolve - derder - c_var - t_DerDer
resolve - DerDer::g - c_function - t_void
resolve - derder - c_var - t_DerDer
resolve - DerDer::h - c_function - t_float
resolve - derder - c_var - t_DerDer
resolve - DerDer::x - UNRESOLVED
